authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -118,27 +127,372 @@ public List<String> policyResourceDefinesRule() {
+    if (RangerHelper.RESOURCE_ALL.equals(authzMetadataObject.name())) {
+      // Remove all schema in this catalog
+      NameIdentifier[] catalogs =
+          GravitinoEnv.getInstance().catalogDispatcher().listCatalogs(Namespace.of(metalake));
+      Arrays.asList(catalogs).stream()
+          .forEach(
+              catalog -> {
+                List<String> catalogLocations =
+                    AuthorizationUtils.getMetadataObjectLocation(
+                        NameIdentifier.of(catalog.name()), Entity.EntityType.CATALOG);
+                catalogLocations.stream()
+                    .forEach(
+                        locationPath -> {
+                          AuthorizationMetadataObject catalogMetadataObject =
+                              new PathBasedMetadataObject(
+                                  metalake, catalog.name(), locationPath, PATH);
+                          doRemoveSchemaMetadataObject(catalogMetadataObject);
+                        });
+              });
+    } else {
+      // Remove all table in this schema
+      NameIdentifier[] tables =
+          GravitinoEnv.getInstance()
+              .tableDispatcher()
+              .listTables(Namespace.of(authzMetadataObject.name()));
+      Arrays.asList(tables).stream()
+          .forEach(
+              table -> {
+                NameIdentifier identifier =
+                    NameIdentifier.of(authzMetadataObject.name(), table.name());
+                List<String> tabLocations =
+                    AuthorizationUtils.getMetadataObjectLocation(
+                        identifier, Entity.EntityType.TABLE);
+                tabLocations.stream()
+                    .forEach(
+                        locationPath -> {
+                          AuthorizationMetadataObject tableMetadataObject =
+                              new PathBasedMetadataObject(
+                                  authzMetadataObject.name(), table.name(), locationPath, PATH);
+                          doRemoveTableMetadataObject(tableMetadataObject);
+                        });
+                // Remove schema
+                Schema schema =
+                    GravitinoEnv.getInstance()
+                        .schemaDispatcher()
+                        .loadSchema(NameIdentifier.of(authzMetadataObject.name()));
+                List<String> schemaLocations =
+                    AuthorizationUtils.getMetadataObjectLocation(
+                        identifier, Entity.EntityType.SCHEMA);
+                schemaLocations.stream()
+                    .forEach(
+                        locationPath -> {
+                          AuthorizationMetadataObject schemaMetadataObject =
+                              new PathBasedMetadataObject(
+                                  authzMetadataObject.name(), schema.name(), locationPath, PATH);
+                          removePolicyByMetadataObject(schemaMetadataObject);
+                        });
+              });
```

Comment:
I'm not entirely clear on those codes, don't we need to remove all tables and then the schema location?

Reply:
I mean we need to remove all tables in a schema first, after all tables have been, then we can start to remove schema. I noticed that the code will remove the schema after we just remove one of tables in it.

Reply:
I already loop delete all tables under this schema.
```
      // Remove all table in this schema
      NameIdentifier[] tables =
          GravitinoEnv.getInstance()
              .tableDispatcher()
              .listTables(Namespace.of(authzMetadataObject.name()));
```